Feature/fix suffix on bulk rule

marksull opened the pull request.

dalamanster commented:
Thanks. One more thing is happening. If parameters are not set they are translated to URL as None.
This is causing problem with delete bulk request url: 

`https://......./accessrules?category=None&insertBefore=None&insertAfter=None&section=None&bulk=true&filter=ids:........`

This will probably fix it:
Changing this:
```
        if "category" in self.__dict__:
            url = f"{url}category={self.category}&"
        if "insertBefore" in self.__dict__:
            url = f"{url}insertBefore={self.insertBefore}&"
        if "insertAfter" in self.__dict__:
            url = f"{url}insertAfter={self.insertAfter}&"
        if "insertBefore" in self.__dict__ and "insertAfter" in self.__dict__:
            logging.warning("ACP rule has both insertBefore and insertAfter params")
        if "section" in self.__dict__:
            url = f"{url}section={self.section}&"
```

to this 
(self.{attribute} has to exist - other than None)
```
        if "category" in self.__dict__ and self.category:
            url = f"{url}category={self.category}&"
        if "insertBefore" in self.__dict__ and self.insertBefore:
            url = f"{url}insertBefore={self.insertBefore}&"
        if "insertAfter" in self.__dict__ and self.insertAfter:
            url = f"{url}insertAfter={self.insertAfter}&"
        if "insertBefore" in self.__dict__ and "insertAfter" in self.__dict__  and self.insertBefore and self.insertAfter:
            logging.warning("ACP rule has both insertBefore and insertAfter params")
        if "section" in self.__dict__  and self.section:
            url = f"{url}section={self.section}&"
```

marksull commented:
@dalamanster yeah thx. I left the PR open because I had not tested it yet so your feedback was appreciated. I fixed the issue above that you raised. I also implemented paging on the delete records in case the number of rules being deleted caused the length of the URL to exceed the maximum length.
